File: app/src/main/java/com/wipro/android/proficiencyexercise/view/detail/DetailFragment.java
package com.wipro.android.proficiencyexercise.view.detail;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.wipro.android.proficiencyexercise.R;
import com.wipro.android.proficiencyexercise.data.remote.response.Row;
import com.wipro.android.proficiencyexercise.view.base.BaseFragment;

public class DetailFragment extends BaseFragment {

    private TextView desc;
    private ImageView image;
    private Row rows;

    public static DetailFragment newInstance(Row rows) {
        Bundle args = new Bundle();
        args.putSerializable("rows", rows);
        DetailFragment fragment = new DetailFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        rows = (Row) (getArguments() != null ? getArguments().getSerializable("rows") : null);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(R.layout.detail_fragment, container, false);
        desc = view.findViewById(R.id.desc);
        image = view.findViewById(R.id.image);
        setViewData();
        return view;
    }

    private void setViewData() {
        if (rows != null) {
            if (rows.getTitle() != null && !rows.getTitle().isEmpty())
                if (getBaseActivity().getSupportActionBar() != null) {
                    getBaseActivity().getSupportActionBar().setTitle(rows.getTitle());
                }
            if (rows.getDescription() != null && !rows.getDescription().isEmpty())
                desc.setText(rows.getDescription());
            if (rows.getImageHref() != null && !rows.getImageHref().isEmpty())
                Glide.with(getBaseActivity())
                        .load(rows.getImageHref())
                        .into(image);
        }
    }
}

File: app/src/main/java/com/wipro/android/proficiencyexercise/view/detail/DetailFragment.kt
package com.wipro.android.proficiencyexercise.view.detail

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.wipro.android.proficiencyexercise.R
import com.wipro.android.proficiencyexercise.data.remote.response.Row
import com.wipro.android.proficiencyexercise.view.base.BaseFragment

class DetailFragment : BaseFragment() {

    private var desc: TextView? = null
    private var image: ImageView? = null
    private var rows: Row? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        rows = (if (arguments != null) arguments!!.getSerializable("rows") else null) as Row
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        val view = inflater.inflate(R.layout.detail_fragment, container, false)
        desc = view.findViewById(R.id.desc)
        image = view.findViewById(R.id.image)
        setViewData()
        return view
    }

    private fun setViewData() {
        if (rows != null) {
            if (rows!!.title != null && !rows!!.title!!.isEmpty())
                if (baseActivity.supportActionBar != null) {
                    baseActivity.supportActionBar!!.title = rows!!.title
                }
            if (rows!!.description != null && !rows!!.description!!.isEmpty())
                desc!!.text = rows!!.description
            if (rows!!.imageHref != null && !rows!!.imageHref!!.isEmpty())
                Glide.with(baseActivity)
                        .load(rows!!.imageHref)
                        .into(image!!)
        }
    }

    companion object {

        fun newInstance(rows: Row): DetailFragment {
            val args = Bundle()
            args.putSerializable("rows", rows)
            val fragment = DetailFragment()
            fragment.arguments = args
            return fragment
        }
    }
}